src/Cli/Microsoft.TemplateEngine.Cli/Commands/BaseCommand.cs

```diff
@@ -176,11 +178,53 @@ protected void SetupTabularOutputOptions(ITabularOutputCommand command)
 
         }
 
-        private static async Task HandleGlobalOptionsAsync(TArgs args, IEngineEnvironmentSettings environmentSettings, CancellationToken cancellationToken)
+        /// <summary>
+        /// Checks if the template with same short name as <paramref name="command"/> exists, and if so prints the example on how to run the template using dotnet new create.
+        /// </summary>
+        /// <remarks>
+        /// This method uses <see cref="TemplatePackageManager.GetTemplatesAsync(CancellationToken)"/>, however this should not take long as templates normally at least once
+        /// are queried before and results are cached.
+        /// Alternatively we can think of caching template groups early in <see cref="BaseCommand{TArgs}"/> later on.
+        /// </remarks>
+        protected internal static async Task CheckTemplatesWithSubCommandName(
+            TArgs args,
+            TemplatePackageManager templatePackageManager,
+            Command command,
+            CancellationToken cancellationToken)
+        {
+            IReadOnlyList<ITemplateInfo> availableTemplates = await templatePackageManager.GetTemplatesAsync(cancellationToken).ConfigureAwait(false);
+            HashSet<string> matchedAliases = new(StringComparer.OrdinalIgnoreCase);
+            foreach (ITemplateInfo templateInfo in availableTemplates) 
+            {
+                foreach (string alias in command.Aliases)
+                {
+                    string? match = templateInfo.ShortNameList.FirstOrDefault(sn => string.Equals(sn, alias, StringComparison.OrdinalIgnoreCase));
+                    if (!string.IsNullOrEmpty(match))
+                    {
+                        matchedAliases.Add(alias);
+                        //we don't want to probe other aliases on the same template, this will result in duplicate messages.
+                        //normally the commands have a single alias
+                        break;
```

Comment:
Oh - I misunderstood the logic on a first sight. I guess it might benefit by a small shortening - I'd probably initialize `HashSet` with existing alias and then select distinct duplicates - something like:
```
HashSet<string> existingAliases = new(availableTemplates.SelectMany(t => t.ShortNameList) StringComparer.OrdinalIgnoreCase);
var duplicates = command.Aliases.Where(alias => existingAliases.Contains(alias)).Distinct();
```

Or if we want just the first match:
```
string? duplicate = command.Aliases.FirstOrEmpty(alias => existingAliases.Contains(alias));
```

Reply:
It's just a theory case as currently all the commands have just single alias.
The idea is that we report only the first alias that matched.  

Actually thinking on it more, we should rather report only the alias that was used, and this always will be the only one.  Let me change it a bit.